sql/core/src/main/scala/org/apache/spark/sql/catalyst/util/V2ExpressionBuilder.scala

```diff
@@ -397,6 +397,33 @@ class V2ExpressionBuilder(e: Expression, isPredicate: Boolean = false) {
       generateExpression(child).map(v => new V2Extract("WEEK", v))
     case YearOfWeek(child) =>
       generateExpression(child).map(v => new V2Extract("YEAR_OF_WEEK", v))
+    case encrypt: AesEncrypt =>
+      val childrenExpressions = encrypt.children.flatMap(generateExpression(_))
+      if (childrenExpressions.length == encrypt.children.length) {
```

Comment:
not related to this PR, but this pattern has been repeated a lot in this file, it's better to abstract it out.

Reply:
OK. I will create another PR to abstract it out.